hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -169,9 +170,10 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
   lazy val partitionSchema: StructType = StructType(tableSchema.filter(f => partitionFields.contains(f.name)))
 
   /**
-   * All the partition paths
+   * All the partition paths, excludes lazily deleted partitions.
    */
   def getPartitionPaths: Seq[String] = getAllPartitionPaths(spark, table)
+    .filter(!TimelineUtils.getDroppedPartitions(metaClient.getActiveTimeline).contains(_))
 
```

Comment:
Will this trigger the calculation for each partition ?

Reply:
Oh! make senses, thanks for point this, let me fix it.